### ISSUE TITLE
Navigation Block: Merge horizontal and vertical nav into one block

### ISSUE THREAD

**Issue by alaczek** [WordPress/gutenberg#23226]
This might have already been discussed, but I was wondering why horizontal and vertical navigation are separate blocks? Would it make sense to make them two styles of the same block? 

Could we remove the light and dark block styles (which are reproducible with background and text color options anyway - see #22327) and replace them with orientation? Here's a rough mockup of how this could look like:

![nav_block_styles](https://user-images.githubusercontent.com/4550351/84852084-40b34300-b09f-11ea-8e4e-bde7c38e1381.jpg)

**Comment by draganescu** [comment]
The vertical orientation of the navigation bloc is a variation because it also modifies how the block works. For instance the movers for the links in the navigation block are vertical instead of horizontal in the vertical variation. 

That cannot happen by simply adding a class as we need to update an attribute that can then be read and affect how the editor renders the block.

**Comment by alaczek** [comment]
Hmm... that is a bummer if these two cannot be merged into one - once I started playing with FSE and creating different headers this option made the most sense to me UX-wise. 

I also found that the vertical menu is getting somewhat lost when using the `/` command. When I started typing `/nav` only one Navigation block was suggested to me without indication if it was horizontal or vertical (it is the horizontal one).

![image](https://user-images.githubusercontent.com/4550351/84961338-62680500-b147-11ea-81e9-8f91bc5c624b.png)

If we're going to keep two separate navigation blocks, would it make sense to indicate orientation in the block empty state too? (right now it just says "Navigation")

![image](https://user-images.githubusercontent.com/4550351/84961473-ba067080-b147-11ea-977e-e1cd5a12da37.png)
